sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/ConsumerSpEL.java

```diff
@@ -50,15 +50,15 @@
       parser.parseExpression("#consumer.assign(#tp)");
 
   private Method timestampMethod;
-  private boolean hasEventTimestamp = false;
+  private boolean hasRecordTimestamp = false;
 
   public ConsumerSpEL() {
     try {
       timestampMethod = ConsumerRecord.class.getMethod("timestamp", (Class<?>[]) null);
     } catch (NoSuchMethodException | SecurityException e) {
```

Comment:
how about `hasRecordTimeStamp = timestampMethod.getReturnType().equals(Long.TYPE)` here?
That way you can remove lines 78-87.